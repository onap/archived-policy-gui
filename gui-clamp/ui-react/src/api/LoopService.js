/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

export default class LoopService {
	static getLoopNames() {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/getAllNames', { method: 'GET', credentials: 'same-origin' })
			.then(function (response) {
				console.debug("GetLoopNames response received: ", response.status);
				if (response.ok) {
					return response.json();
				} else {
					console.error("GetLoopNames query failed");
					return {};
				}
			})
			.catch(function (error) {
				console.error("GetLoopNames error received", error);
				return {};
			});
	}

	static createLoop(loopName, templateName) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/create/' + loopName + '?templateName=' + templateName, {
			method: 'POST',
			headers: {
				"Content-Type": "application/json"
			},
			credentials: 'same-origin'
		})
			.then(function (response) {
				console.debug("CreateLoop response received: ", response.status);
				return response.json();
			})
			.catch(function (error) {
				console.error("CreateLoop error received", error);
				return "";
			});
	}

	static getLoop(loopName) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/' + loopName, {
			method: 'GET',
			headers: {
				"Content-Type": "application/json"
			},
			credentials: 'same-origin'
		})
			.then(function (response) {
				console.debug("GetLoop response received: ", response.status);
				if (response.ok) {
					return response.json();
				} else {
					console.error("GetLoop query failed");
					return {};
				}
			})
			.catch(function (error) {
				console.error("GetLoop error received", error);
				return {};
			});
	}

	static setMicroServiceProperties(loopName, jsonData) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/updateMicroservicePolicy/' + loopName, {
			method: 'POST',
			credentials: 'same-origin',
			headers: {
				"Content-Type": "application/json"
			},
			body: JSON.stringify(jsonData)
		})
			.then(function (response) {
				console.debug("updateMicroservicePolicy response received: ", response.status);
				if (response.ok) {
					return response.text();
				} else {
					console.error("updateMicroservicePolicy query failed");
					return "";
				}
			})
			.catch(function (error) {
				console.error("updateMicroservicePolicy error received", error);
				return "";
			});
	}
	
	static setOperationalPolicyProperties(loopName, jsonData) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/updateOperationalPolicies/' + loopName, {
			method: 'POST',
			credentials: 'same-origin',
			headers: {
				"Content-Type": "application/json"
			},
			body: JSON.stringify(jsonData)
		})
			.then(function (response) {
				console.debug("updateOperationalPolicies response received: ", response.status);
				if (response.ok) {
					return response.text();
				} else {
					console.error("updateOperationalPolicies query failed");
					return "";
				}
			})
			.catch(function (error) {
				console.error("updateOperationalPolicies error received", error);
				return "";
			});
	}

	static updateGlobalProperties(loopName, jsonData) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/updateGlobalProperties/' + loopName, {
			method: 'POST',
			credentials: 'same-origin',
			headers: {
				"Content-Type": "application/json"
			},
			body: JSON.stringify(jsonData)
		})
			.then(function (response) {
				console.debug("updateGlobalProperties response received: ", response.status);
				if (response.ok) {
					return response.text();
				} else {
					console.error("updateGlobalProperties query failed");
					return "";
				}
			})
			.catch(function (error) {
				console.error("updateGlobalProperties error received", error);
				return "";
			});
	}

	static refreshOperationalPolicyJson(loopName,operationalPolicyName) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/refreshOperationalPolicyJsonSchema/' + loopName + '/' + operationalPolicyName, {
			method: 'PUT',
			headers: {
				"Content-Type": "application/json"
			},
			credentials: 'same-origin'
		})
			.then(function (response) {
				console.debug("Refresh Operational Policy Json Schema response received: ", response.status);
				if (response.ok) {
					return response.json();
				} else {
					console.error("Refresh Operational Policy Json Schema query failed");
					return {};
				}
			})
			.catch(function (error) {
				console.error("Refresh Operational Policy Json Schema error received", error);
				return {};
			});
	}

		static refreshMicroServicePolicyJson(loopName,microServicePolicyName) {
    		return fetch(window.location.pathname + 'restservices/clds/v2/loop/refreshMicroServicePolicyJsonSchema/' + loopName + '/' + microServicePolicyName, {
    			method: 'PUT',
    			headers: {
    				"Content-Type": "application/json"
    			},
    			credentials: 'same-origin'
    		})
    			.then(function (response) {
    				console.debug("Refresh Operational Policy Json Schema response received: ", response.status);
    				if (response.ok) {
    					return response.json();
    				} else {
    					console.error("Refresh Operational Policy Json Schema query failed");
    					return {};
    				}
    			})
    			.catch(function (error) {
    				console.error("Refresh Operational Policy Json Schema error received", error);
    				return {};
    			});
    }

	static addOperationalPolicyType(loopName, policyType, policyVersion) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/addOperationaPolicy/' + loopName + '/policyModel/' + policyType +'/' + policyVersion , {
			method: 'PUT',
			headers: {
				"Content-Type": "application/json"
			},
			credentials: 'same-origin'
		})
				.then(function (response) {
				console.debug("Add Operational Policy response received: ", response.status);
				if (response.ok) {
					return response.json();
				} else {
					return response.text();
				}
			})
			.catch(function (error) {
				console.error("Add Operational Policy query failed");
				throw new Error(error);
			})
	}

	static removeOperationalPolicyType(loopName, policyType, policyVersion, policyName) {
		return fetch(window.location.pathname + 'restservices/clds/v2/loop/removeOperationaPolicy/' + loopName + '/policyModel/' + policyType +'/' + policyVersion + '/' + policyName , {
			method: 'PUT',
			headers: {
				"Content-Type": "application/json"
			},
			credentials: 'same-origin'
		})
				.then(function (response) {
					console.debug("Remove Operational Policy response received: ", response.status);
				if (response.ok) {
					return response.json();
				} else {
					console.error("Remove Operational Policy query failed");
					return {};
				}
			})
			.catch(function (error) {
				console.error("Remove Operational Policy error received", error);
				return {};
			});
	}
}
