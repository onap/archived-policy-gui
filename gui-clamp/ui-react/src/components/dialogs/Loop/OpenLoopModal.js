/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019 AT&T Intellectual Property. All rights reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */

import React from 'react'
import Select from 'react-select';
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';
import Form from 'react-bootstrap/Form';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import FormCheck from 'react-bootstrap/FormCheck'
import styled from 'styled-components';
import LoopService from '../../../api/LoopService';
import SvgGenerator from '../../loop_viewer/svg/SvgGenerator';
import LoopCache from '../../../api/LoopCache';

const ModalStyled = styled(Modal)`
	background-color: transparent;
`
const CheckBoxStyled = styled(FormCheck.Input)`
	margin-left:3rem;
`

export default class OpenLoopModal extends React.Component {
	constructor(props, context) {
		super(props, context);

		this.getLoopNames = this.getLoopNames.bind(this);
		this.handleOpen = this.handleOpen.bind(this);
		this.handleClose = this.handleClose.bind(this);
		this.handleDropDownListChange = this.handleDropDownListChange.bind(this);
		this.renderSvg = this.renderSvg.bind(this);
		this.showReadOnly = props.showReadOnly !== undefined ? props.showReadOnly : true;
		this.state = {
			show: true,
			chosenLoopName: '',
			loopNames: [],
			loopCacheOpened: new LoopCache({})
		};
	}

	componentWillMount() {
		this.getLoopNames();
	}

	handleClose() {
		this.setState({ show: false });
		this.props.history.push('/');
	}

	handleDropDownListChange(e) {
        LoopService.getLoop(e.value).then(loop => {
            this.setState({
                chosenLoopName: e.value,
                loopCacheOpened: new LoopCache(loop)
             });
		});
	}

	getLoopNames() {
		LoopService.getLoopNames().then(loopNames => {
		    if (Object.entries(loopNames).length !== 0) {
		        const loopOptions = loopNames.filter(loopName => loopName!=='undefined').map((loopName) => { return { label: loopName, value: loopName } });
            	this.setState({ loopNames: loopOptions })
		    }
		});
	}

	handleOpen() {
		console.info("Loop " + this.state.chosenLoopName + " is chosen");
        this.handleClose();
		this.props.loadLoopFunction(this.state.chosenLoopName);
	}

	renderSvg() {
		return(
				<SvgGenerator loopCache={this.state.loopCacheOpened} clickable={false} generatedFrom={SvgGenerator.GENERATED_FROM_INSTANCE}/>
		);
	}

	render() {
		return (
			<ModalStyled size="xl" show={this.state.show} onHide={this.handleClose} backdrop="static" keyboard={false} >
				<Modal.Header closeButton>
					<Modal.Title>Open Model</Modal.Title>
				</Modal.Header>
				<Modal.Body>
					<Form.Group as={Row} controlId="formPlaintextEmail">
						<Form.Label column sm="2">Model Name:</Form.Label>
						<Col sm="10">
							<Select onChange={this.handleDropDownListChange}
							options={this.state.loopNames} />
						</Col>
					</Form.Group>
					<Form.Group as={Row} style={{alignItems: 'center'}} controlId="formSvgPreview">
						<Form.Label column sm="2">Model Preview:</Form.Label>
						<Col sm="10">
						    {this.renderSvg()}
						</Col>
					</Form.Group>
					{this.showReadOnly === true ?
						<Form.Group as={Row} controlId="formBasicCheckbox">
							<Form.Check>
								<FormCheck.Label>Read Only Mode:</FormCheck.Label>
								<CheckBoxStyled style={{marginLeft: '3.5em'}} type="checkbox" />
							</Form.Check>
						</Form.Group>
					: null}
				</Modal.Body>
				<Modal.Footer>
					<Button variant="secondary" type="null" onClick={this.handleClose}>Cancel</Button>
					<Button variant="primary" type="submit" onClick={this.handleOpen}>Open</Button>
				</Modal.Footer>
			</ModalStyled>

		);
	}
}
