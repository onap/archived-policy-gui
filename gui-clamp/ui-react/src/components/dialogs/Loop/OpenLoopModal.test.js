/*-
 * ============LICENSE_START=======================================================
 * ONAP CLAMP
 * ================================================================================
 * Copyright (C) 2019 AT&T Intellectual Property. All rights
 *                             reserved.
 * ================================================================================
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ============LICENSE_END============================================
 * ===================================================================
 *
 */
import React from 'react';
import { shallow } from 'enzyme';
import OpenLoopModal from './OpenLoopModal';
import LoopService from '../../../api/LoopService';

describe('Verify OpenLoopModal', () => {

  beforeEach(() => {
      fetch.resetMocks();
      fetch.mockResponse(JSON.stringify([
        "LOOP_gmtAS_v1_0_ResourceInstanceName1_tca",
        "LOOP_gmtAS_v1_0_ResourceInstanceName1_tca_3",
        "LOOP_gmtAS_v1_0_ResourceInstanceName2_tca_2"
      ]));
  });

    it('Test the render method', () => {

    const component = shallow(<OpenLoopModal/>);
    expect(component).toMatchSnapshot();
  });

  it('Onchange event', async () => {
    const flushPromises = () => new Promise(setImmediate);
    LoopService.getLoop = jest.fn().mockImplementation(() => {
  		return Promise.resolve({
  			ok: true,
  			status: 200,
  			json: () => {}
  		});
  	});
    const event = {value: 'LOOP_gmtAS_v1_0_ResourceInstanceName1_tca_3'};
    const component = shallow(<OpenLoopModal/>);
    component.find('StateManager').simulate('change', event);
    await flushPromises();
    component.update();
    expect(component.state('chosenLoopName')).toEqual("LOOP_gmtAS_v1_0_ResourceInstanceName1_tca_3");
  });


  it('Test handleClose', () => {
    const historyMock = { push: jest.fn() }; 
    const handleClose = jest.spyOn(OpenLoopModal.prototype,'handleClose');
    const component = shallow(<OpenLoopModal history={historyMock} />)

    component.find('[variant="secondary"]').prop('onClick')();

    expect(handleClose).toHaveBeenCalledTimes(1);
    expect(component.state('show')).toEqual(false);
    expect(historyMock.push.mock.calls[0]).toEqual([ '/']);
    
    handleClose.mockClear();
  });

    it('Test handleSubmit', () => {
    const historyMock = { push: jest.fn() };
    const loadLoopFunction = jest.fn();  
    const handleOpen = jest.spyOn(OpenLoopModal.prototype,'handleOpen');
    const component = shallow(<OpenLoopModal history={historyMock} loadLoopFunction={loadLoopFunction}/>)

    component.find('[variant="primary"]').prop('onClick')();

    expect(handleOpen).toHaveBeenCalledTimes(1);
    expect(component.state('show')).toEqual(false);
    expect(historyMock.push.mock.calls[0]).toEqual([ '/']);

    handleOpen.mockClear();
  });

});
