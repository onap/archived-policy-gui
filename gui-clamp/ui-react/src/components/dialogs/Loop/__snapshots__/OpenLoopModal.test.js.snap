// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify OpenLoopModal Test the render method 1`] = `
<Styled(Modal)
  backdrop="static"
  keyboard={false}
  onHide={[Function]}
  show={true}
  size="xl"
>
  <ModalHeader
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle>
      Open Model
    </ModalTitle>
  </ModalHeader>
  <ModalBody>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "noGutters": false,
          },
          "render": [Function],
        }
      }
      controlId="formPlaintextEmail"
    >
      <FormLabel
        column={true}
        sm="2"
        srOnly={false}
      >
        Model Name:
      </FormLabel>
      <Col
        sm="10"
      >
        <StateManager
          defaultInputValue=""
          defaultMenuIsOpen={false}
          defaultValue={null}
          onChange={[Function]}
          options={Array []}
        />
      </Col>
    </FormGroup>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "noGutters": false,
          },
          "render": [Function],
        }
      }
      controlId="formSvgPreview"
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <FormLabel
        column={true}
        sm="2"
        srOnly={false}
      >
        Model Preview:
      </FormLabel>
      <Col
        sm="10"
      >
        <withRouter(SvgGenerator)
          clickable={false}
          generatedFrom="INSTANCE"
          loopCache={
            LoopCache {
              "loopJsonCache": Object {},
            }
          }
        />
      </Col>
    </FormGroup>
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "noGutters": false,
          },
          "render": [Function],
        }
      }
      controlId="formBasicCheckbox"
    >
      <FormCheck>
        <FormCheckLabel>
          Read Only Mode:
        </FormCheckLabel>
        <Styled(FormCheckInput)
          style={
            Object {
              "marginLeft": "3.5em",
            }
          }
          type="checkbox"
        />
      </FormCheck>
    </FormGroup>
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="null"
      variant="secondary"
    >
      Cancel
    </Button>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="submit"
      variant="primary"
    >
      Open
    </Button>
  </ModalFooter>
</Styled(Modal)>
`;
